Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -372,4 +373,131 @@ final class DocumentTests: XCTestCase {
         expect(doc["a1"]).to(equal(arr1))
         expect(doc["a2"]).to(equal(arr2))
     }
+
+    // exclude Int64 value on 32-bit platforms
+    static let overwritables: Document = XCTestCase.is32Bit ?
+                                        ["double": 2.5, "int32": Int32(32), "bool": false, "decimal": Decimal128("1.2E+10")] :
+                                        ["double": 2.5, "int32": Int32(32), "int64": Int64(64), "bool": false, "decimal": Decimal128("1.2E+10")]
+
+    static let nonOverwritables: Document = ["string": "hello", "nil": nil, "doc": ["x": 1] as Document, "arr": [1, 2] as [Int], "oid": ObjectId()]
+
+    // test replacing `Overwritable` types with values of their own type
```

Comment:
If I understand correctly, this test ovewrites values but doesn't copy the underlying storage, since libbson lets us overwrite them in place. But we also have tests for other overwriting behavior:

 * test replacing some of the non-Overwritable types with values of their own types
 * test replacing both overwritable and nonoverwritable values with values of different types

I wonder if _this_ test should actually assert that we don't duplicate storage. If so, perhaps that other tests can assert that storage _is_ duplicated (where appropriate).